numpy/_core/fromnumeric.py

```diff
@@ -2643,6 +2665,202 @@ def all(a, axis=None, out=None, keepdims=np._NoValue, *, where=np._NoValue):
+            raise ValueError("For arrays which have more than one dimension "
+                            "``axis`` argument is required.")
+        axis = 0
+
+    if out is not None and include_initial:
+        item = [slice(None)] * x_ndim
+        item[axis] = slice(1, None)
+        func.accumulate(x, axis=axis, dtype=dtype, out=out[tuple(item)])
+        item[axis] = 0
+        out[tuple(item)] = func.identity
+        return out
+
+    res = func.accumulate(x, axis=axis, dtype=dtype, out=out)
+    if include_initial:
+        initial_shape = list(x.shape)
+        initial_shape[axis] = 1
+        res = np.concat(
+            [np.full_like(res, func.identity, shape=initial_shape), res],
+            axis=axis,
+        )
+
+    return res
+
+
+def _cumulative_prod_dispatcher(x, /, *, axis=None, dtype=None, out=None,
+                                include_initial=None):
+    return (x, out)
+
+
+@array_function_dispatch(_cumulative_prod_dispatcher)
+def cumulative_prod(x, /, *, axis=None, dtype=None, out=None,
+                    include_initial=False):
+    """
+    Return the cumulative product of elements along a given axis.
+
+    This function is an Array API compatible alternative to `numpy.cumprod`.
+
+    Parameters
+    ----------
+    x : array_like
+        Input array.
+    axis : int, optional
+        Axis along which the cumulative product is computed. The default
+        (None) is only allowed for one-dimensional arrays. For arrays
+        with more than one dimension ``axis`` is required.
+    dtype : dtype, optional
+        Type of the returned array, as well as of the accumulator in which
+        the elements are multiplied.  If ``dtype`` is not specified, it
+        defaults to the dtype of ``x``, unless ``x`` has an integer dtype
+        with a precision less than that of the default platform integer.
+        In that case, the default platform integer is used instead.
+    out : ndarray, optional
+        Alternative output array in which to place the result. It must
+        have the same shape and buffer length as the expected output
+        but the type of the resulting values will be cast if necessary.
+        See :ref:`ufuncs-output-type` for more details.
+    include_initial : bool, optional
+        Boolean indicating whether to include the initial value (ones) as
+        the first value in the output. With ``include_initial=True``
+        the shape of the output is different than the shape of the input.
+        Default: ``False``.
+
+    Returns
+    -------
+    cumulative_prod_along_axis : ndarray
+        A new array holding the result is returned unless ``out`` is
+        specified, in which case a reference to ``out`` is returned. The
+        result has the same shape as ``x`` if ``include_initial=False``.
+
+    Notes
+    -----
+    Arithmetic is modular when using integer types, and no error is
+    raised on overflow.
+
+    Examples
+    --------
+    >>> a = np.array([1,2,3])
```

Comment:
While you are at it, one more space after comma...

Reply:
Done!